### PR TITLE
Get call to action for chat permissions

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Paves the way for the new UI describing why you can't message someone, by adding a new selector that infers the reason based on the permissions for a user.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/common/src/store/pages/chat/selectors.ts [hunk 1]
```diff
@@ -150,3 +151,62 @@ export const getUnfurlMetadata = (
   const message = getChatMessageById(state, chatId, messageId)
   return message?.unfurlMetadata
 }
+
+export const getUserChatPermissions = createSelector(
```
Comment (rickyrombo) [inline packages/common/src/store/pages/chat/selectors.ts:155]: fyi: @dharit-tan breaking change for mobile potentially

File: packages/common/src/store/pages/chat/selectors.ts [hunk 2]
```diff
@@ -150,3 +151,62 @@ export const getUnfurlMetadata = (
...
+    const currentUserPermissions = currentUserId
+      ? chatPermissions[currentUserId]
+      : null
+    const userPermissions = chatPermissions[userId]
+    const isBlockee = blockees.includes(userId)
+    const isBlocker = blockers.includes(userId)
+    const canChat =
+      !isBlockee &&
+      !isBlocker &&
+      (userPermissions?.current_user_has_permission ?? true)
```
Comment (michellebrier) [inline packages/common/src/store/pages/chat/selectors.ts:188]: this wont be false bidirectionally yet, so the logic below that calcs actions based on the `currentUserPermissions` gated with (`!canChat)` wont be evaluated if the non-current user is not a blocker/blockee and has permissions such that the current user is allowed to chat with them. 

tldr we need a bidirectional `current_user_has_permission` for this to work, and we can calc the action on the backend or the frontend but would prefer to just keep it all contained in the backend (and frontend reads a `to_enable` field in the resp)

Comment (rickyrombo) [reply packages/common/src/store/pages/chat/selectors.ts:188]: ahh yeah ok that checks out. Thanks!

File: packages/common/src/store/pages/chat/selectors.ts (hunk 1 above)
Comment (dharit-tan) [reply packages/common/src/store/pages/chat/selectors.ts:155]: Love this

File: packages/common/src/store/pages/chat/selectors.ts (hunk 2 above)
Comment (rickyrombo) [reply packages/common/src/store/pages/chat/selectors.ts:188]: @michellebrier / @dharit-tan does this work now that we have unidirectional permissions?

Comment (dharit-tan) [reply packages/common/src/store/pages/chat/selectors.ts:188]: Think so

**Review by dharit-tan — APPROVED** [review]

**Comment by dharit-tan** [comment]
Lint + merge main!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-permissions-action

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-permissions-action
